app/controllers/api/base_controller.rb

```diff
@@ -231,6 +232,20 @@ def not_found_if_nested_id_exists
       end
     end
 
+    def missing_permissions
+      Foreman::AccessControl.permissions_for_controller_action(path_to_authenticate)
+    end
+
+    def set_error_details(error, options)
+      case error
+      when 'access_denied'
+        if options.is_a?(Hash) && options.fetch(:locals, {}).fetch(:details, nil).blank?
+          options = options.deep_merge({:locals => {:details => _('Missing one of the required permissions: %s') % missing_permissions.map(&:name).join(', ') }})
```

Comment:
I think this method can be simplified by simply returning the error details, and making `render_error` just

```ruby
options.deep_merge(:locals => { :details => error_details(error) })
```